data-connection/src/main/java/io/micronaut/data/connection/ConnectionDefinition.java

```diff
@@ -101,6 +102,15 @@ enum Propagation {
     @Nullable
     String getName();
 
+    /**
+     * Returns the connection client information associated with this connection definition.
+     * If no connection client information has been set, this method will return null.
+     *
+     * @return An instance of {@link ConnectionClientInfoDetails} representing the connection client information, or null if not set.
+     * @since 4.10
+     */
+    @Nullable ConnectionClientInfoDetails connectionClientInfo();
```

Comment:
Please remove this; let it be fetched by the Oracle listener. Make `ConnectionDefinition` `AnnotationMetadataProvider`